ext/xmlreader/php_xmlreader.c

```diff
@@ -123,6 +123,25 @@ zval *xmlreader_get_property_ptr_ptr(zend_object *object, zend_string *name, int
 }
 /* }}} */
 
+static int xmlreader_has_property(zend_object *object, zend_string *name, int type, void **cache_slot)
+{
+	xmlreader_object *obj = php_xmlreader_fetch_object(object);
+	xmlreader_prop_handler *hnd = zend_hash_find_ptr(&xmlreader_prop_handlers, name);
+
+	if (hnd != NULL) {
+		zval rv;
+		if (xmlreader_property_reader(obj, hnd, &rv) == FAILURE) {
+			return 0;
+		} else {
+			zval_ptr_dtor(&rv);
+			return 1;
```

Comment:
This should respect the rules for isset and empty, so for empty this should call `zend_is_true` for example.

Reply:
Ah, yeah, I wanted to copy-paste [the code I wrote for DatePeriod](https://github.com/php/php-src/pull/15598/files#diff-975ac482adc74b487c6ac9d8dde32153e4b8a803ef8ec41e8d6594f0a042db69R5962), but somehow I got distracted... I'm fixing it right now.